en/companion_computer/pixhawk_rpi.md

```diff
@@ -0,0 +1,259 @@
+
+https://ubuntu.com/tutorials/how-to-install-ubuntu-desktop-on-raspberry-pi-4#1-overview
+
+Next, connect the SD card in the RPi. Connect the mouse, keyboard, monitor and connect the RPi on 5v Power Supply (external source/charger).
+
+Turn on the RPi to boot on SD card and install Ubuntu. In the sequence, install raspi-config in the RPi's terminal:
+
+```
+sudo apt update
+sudo apt upgrade
+sudo apt-get install raspi-config 
+```
+
+Open the raspi-config:
+
+```
+sudo raspi-config
+```
+
+Go to the **Interface Option** and then click at **Serial Port**. Select **No** to disable serial login shell and select **Yes** to enable serial interface. Click **finish** and restart the RPi.
+
+In the sequence, in the RPi's terminal:
+
+```
+sudo nano /boot/firmware/config.txt
+```
+
+Include the following, after the last line:
+
+```
+enable_uart=1
+dtoverlay=disable-bt
+```
+
+Then, **ctrl+x**, **ctrl+y**, **Enter** and restart the RPi.
+
+To check if the serial port is available, in the RPi's terminal, confirm the list of serial devices by typing:
+
+```
+cd /
+ls /dev/ttyAMA0
+```
+
+On RPi, the default name of a RX/TX connection is `/dev/ttyAMA0`. The same serial port is also available as `/dev/serial0`.
+
+Next, install the mavproxy in the RPi's terminal:
+
+```
+sudo apt install python3-pip
+sudo pip3 install mavproxy
+sudo apt remove modemmanager
+```
+
+## MAVlink communication
+
+Connect the Pixhawk with the laptop via `USB` cable and change the following parameters in QGroundControl:
+
+```
+MAV_0_CONFIG = TELEM2
+XRCE_DDS_0_CFG = Disabled
+SER_TEL1_BAUD = 57600
+```
+
+Next, connect the Pixhawk `TELEM2` pins TX/RX/Ground on RPi correspondent pins. We are going to leave the `TELEM1` port for a setup with a Radio Controller.
+
+The connections listed below must be done:
+
+1. `TELEM2` TX -> RPi RXD (GPIO 15 - pin 10)
+2. `TELEM2` RX -> RPi TXD (GPIO 14 - pin 8)
+3. `TELEM2` GND -> RPi Ground (pin 6)
+
+It is important to identify the correct pins in the hardware. The image illustrates the `TELEM2` pins in the Pixhawk-6C Model.
+
+<img src="../../assets/companion_computer/pixhawk_rpi/pixhawk_pins.png" width="914" height="360" /> 
+
+Ps.: The pins are numbered according to the following:
+
+<img src="../../assets/companion_computer/pixhawk_rpi/pins_numbers.png" width="615" height="232" /> 
+
+The following image illustrates the correspondent pins at the RPi GPIO.
+
+<img src="../../assets/companion_computer/pixhawk_rpi/rpi_gpio.png" width="845" height="457" /> 
+
+To supply the Pixhawk with 5 Volts is possible to follow with the connection of `POWER1` port with a LiPO battery, via Power Module (PM02, for example). Another option, for this setup moment, is to supply the 5 Volts via the connection of the Pixhawk's `USB` with a smartphone charger, for example, or by keeping it in the laptop `USB`. 
+
+Also, for this setup moment, the RPi is charged with 5 Volts via Power Supply (external source/charger). 
```

Comment:
Fair enough. What are you thinking for the real drone? 5V BEC? Separate battery?

Reply:
@hamishwillee This is a very good question. Let me tell you that I connect the `TELEM2` VCC to supply the RPi in the 5V power (pin 4). 

I am telling in tutorial that this is not recommended because I already read about it. Since I am not really confident about the risks to damage the Pixhawk, I did not write it in the tutorial. But in my case, I have never had any problems with this setup.

Reply:
This is definitely risky. None of the ports on Pixhawk are rated for very high power supply. Generally they supply enough for an old-style 3DR telemetry radio, but if you're attaching a newer high power radio like a microhard setup then you need to power it separately.

What this means is that so far you have been lucky-ish. The RaPi you are using is not drawing that much more power than the old telemetry radio. 

So recommendation would definitely be a BEC. I'll tidy this up when I edit.

Reply:
I fully agree with you, and yes, I believe I have been lucky-ish  :-)